### PR TITLE
Pin pulumi to Python 3.11

### PR DESCRIPTION
**Pull request by tcnghia**
Current grpcio dep version doesn't work on 3.12.

### CODE REVIEW

File: images/pulumi/configs/latest.apko.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@ contents:
     - aspnet-7-targeting-pack
     # for Pulumi Python support
     - pulumi-language-python
-    - python3
-    - python3-dev
-    - py3-pip
+    - python-3.11
```
Comment (k4leung4) [inline images/pulumi/configs/latest.apko.yaml:18]: add a comment to explain why its pinned to 3.11

Comment (tcnghia) [inline images/pulumi/configs/latest.apko.yaml:18]:
```suggestion
    # Currently pulumi has a dependency on grpcio version 1.56, which does not work with 3.11
    # See https://github.com/grpc/grpc/issues/33063.
    - python-3.11
```

File: images/pulumi/configs/latest.apko.yaml [hunk 2]
```diff
@@ -15,9 +15,11 @@ contents:
     - aspnet-7-targeting-pack
     # for Pulumi Python support
     - pulumi-language-python
-    - python3
-    - python3-dev
-    - py3-pip
+    # Currently pulumi has a dependency on grpcio version 1.56, which does not work with 3.11
```
Comment (k4leung4) [inline images/pulumi/configs/latest.apko.yaml:18]: did you mean not work for 3.12?

Comment (tcnghia) [inline images/pulumi/configs/latest.apko.yaml:18]:
```suggestion
    # Currently pulumi has a dependency on grpcio version 1.56, which does not work with 3.12
```

File: images/pulumi/configs/latest.apko.yaml (hunk 1 above)
Comment (tcnghia) [reply images/pulumi/configs/latest.apko.yaml:20]: Done

**Review by k4leung4 — APPROVED** [review]

**Comment by tcnghia** [comment]
@k4leung4 I'm waiting to hear from @kaniini first to merge. Thanks for the review!

**Pull request closed without merging**